application-admintools-default/src/main/java/com/xwiki/admintools/internal/data/ConfigurationDataProvider.java

```diff
@@ -0,0 +1,183 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.admintools.internal.data;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.xwiki.component.annotation.Component;
+
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+import com.xwiki.admintools.ServerIdentifier;
+import com.xwiki.admintools.internal.data.identifiers.CurrentServer;
+import com.xwiki.admintools.internal.util.DefaultFileOperations;
+
+/**
+ * Extension of {@link AbstractDataProvider} for retrieving configuration data.
+ *
+ * @version $Id$
+ * @since 1.0
+ */
+@Component
+@Named(ConfigurationDataProvider.HINT)
+@Singleton
+public class ConfigurationDataProvider extends AbstractDataProvider
+{
+    /**
+     * The hint for the component.
+     */
+    public static final String HINT = "CONFIGURATION";
+
+    private final String template = "configurationTemplate.vm";
+
+    @Inject
+    private DefaultFileOperations fileOperations;
+
+    @Inject
+    private CurrentServer currentServer;
+
+    @Override
+    public String getIdentifier()
+    {
+        return HINT;
+    }
+
+    @Override
+    public String getRenderedData()
+    {
+        Map<String, String> systemInfo = new HashMap<>();
+        try {
+            systemInfo = getDataAsJSON();
+            systemInfo.put(SERVER_FOUND, "true");
+        } catch (Exception e) {
+            systemInfo.put(SERVER_FOUND, "false");
+        }
+        return renderTemplate(this.template, systemInfo, HINT.toLowerCase());
+    }
+
+    @Override
+    public Map<String, String> getDataAsJSON() throws Exception
+    {
+        try {
+            Map<String, String> systemInfo = new HashMap<>();
+            systemInfo.put("database", this.identifyDB());
+            systemInfo.put("xwikiCfgPath", getCurrentServer().getXwikiCfgFolderPath());
+            systemInfo.put("tomcatConfPath", this.getCurrentServer().getServerCfgPath());
+            systemInfo.put("javaVersion", this.getJavaVersion());
+            systemInfo.put("usedServer", this.getCurrentServer().getComponentHint());
+            systemInfo.put("xwikiVersion", getXWikiVersion());
+            systemInfo.putAll(this.getOSInfo());
+            return systemInfo;
+        } catch (Exception e) {
+            throw new Exception(String.format("Failed to generate the instance configuration data. Traceback error: %s",
+                ExceptionUtils.getRootCauseMessage(e)));
```

Comment:
What I suggested is to use
`throw new Exception("Failed to generate the instance configuration data.", e);`
I left the String.format by mistake

Reply:
If I use this format, it does not show the exception received.